hld/approval/correlator.go

```diff
@@ -0,0 +1,206 @@
+package approval
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	humanlayer "github.com/humanlayer/humanlayer/humanlayer-go"
+)
+
+// MemoryStore is an in-memory implementation of Store
```

Comment:
you can enforce this in code with the following, should be using this pattern everywhere
```suggestion
// MemoryStore is an in-memory implementation of Store
var _ Store = &MemoryStore{}
```